src/Modules/OrchardCore.Commerce/Events/WorkflowShoppingCartEvents.cs

```diff
@@ -91,15 +90,42 @@ private T GetOutput<T>(IEnumerable<WorkflowExecutionContext> contexts, string ou
             .SelectWhere(context => context.Output.GetMaybe(outputName))
             .FirstOrDefault();
 
-        return output switch
+        if (output is string text && typeof(T) == typeof(SerializableLocalizedHtmlString))
+        {
+            return Localize(text) as T;
+        }
+
+        var node = output switch
         {
-            string outputLocalizedHtmlString when typeof(T) == typeof(LocalizedHtmlString) =>
-                new LocalizedHtmlString(outputLocalizedHtmlString, outputLocalizedHtmlString) as T,
-            string outputString => JsonConvert.DeserializeObject<T>(outputString, _settings),
-            { } outputObject => JsonConvert.DeserializeObject<T>(
-                JsonConvert.SerializeObject(outputObject, _settings),
-                _settings),
-            _ => default,
+            string json => JNode.Parse(json),
+            not null => JNode.FromObject(output),
+            _ => null,
         };
+
+        if (node is null) return default;
```

Comment:
FYI, in OC we prefer using `{}` even with one line code.

Reply:
I know. In OCC and Lombiq we permit one liners if they are short and trivial, especially with returning null, default, single variable or with throwing an exception.